src/redux/movies.ts

```diff
@@ -0,0 +1,121 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import type { PayloadAction } from '@reduxjs/toolkit';
+import { Movie } from 'types/movie';
+import { RootState } from './store';
+
+interface MoviesState {
+  movies: Movie[];
+  selectedMovie?: Movie;
+  moviesCount: number;
+}
+interface fetchProps {
```

Comment:
1. Not really agree with name: in this case you are using not component props, but allowed search params. So, you can just call it `MoviesSearchParams` or `MoviesSearchOptions` (`Movies` is also applicable here since there might be search also by other entities :))
2. Interfaces should be in _PascalCase_ as same as type aliases